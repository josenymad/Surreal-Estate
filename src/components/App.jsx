import React, { useState, useEffect } from "react";
import "../styles/app.css";
import { Route, Routes } from "react-router-dom";
import { googleLogout, useGoogleLogin } from "@react-oauth/google";
import getProfile from "../requests/getProfile";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import Alert from "./Alert";
import logo from "../images/logo.jpeg";

const App = () => {
  const [user, setUser] = useState([]);
  const [profile, setProfile] = useState([]);
  const [alert, setAlert] = useState({ message: "", isSuccess: false });

  useEffect(() => {
    getProfile(user, setProfile, setAlert);
  }, [user]);

  const logIn = useGoogleLogin({
    onSuccess: (codeResponse) => setUser(codeResponse),
    onError: (error) =>
      setAlert({
        message: `Sorry, could not log in due to this error: ${error}`,
        isSuccess: false,
      }),
  });

  const logOut = () => {
    googleLogout();
    setProfile(null);
  };

  return (
    <div className="App">
      <div className="App__banner">
        <img src={logo} alt="Logo" className="App__header__logo" />
        <h1 className="App__header">Surreal Estate</h1>
      </div>
      {profile ? (
        <div className="App__logged-in">
          <div className="App__logged-in-container">
            <div className="App__account">
              <p className="App__profile-name">Signed in as {profile.name}</p>
              <button
                className="App__log-out__button"
                onClick={logOut}
                type="button"
              >
                Log out
              </button>
            </div>
            <img
              src={profile ? profile.picture : ""}
              alt="profile"
              className="profile-picture"
            />
            <NavBar />
          </div>
          <Routes>
            <Route path="/" element={<Properties />} />
            <Route path="/add-property" element={<AddProperty />} />
          </Routes>
        </div>
      ) : (
        <div className="App__logged-out">
          <button
            className="App__log-in__button"
            onClick={() => logIn()}
            type="button"
          >
            Google Login
          </button>
          <Alert message={alert.message} success={alert.isSuccess} />
        </div>
      )}
    </div>
  );
};

export default App;
