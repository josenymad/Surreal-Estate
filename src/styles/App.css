.App {
  text-align: center;
}

.App__banner {
  display: flex;
  justify-content: center;
  align-items: center;
}

.App__header {
  margin-left: 0.6em;
}

.App__header__logo {
  height: 60px;
  width: auto;
}

.App__logged-in-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  margin: 1em;
  align-items: center;
}

.profile-picture {
  border-radius: 60px;
  justify-self: center;
}

.App__log-out__button, .App__log-in__button {
  padding: 0.6em;
  font-weight: 600;
  background-color: black;
  color: white;
  border-style: none;
  border-radius: 3px;
}