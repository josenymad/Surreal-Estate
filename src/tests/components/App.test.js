import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { GoogleOAuthProvider } from "@react-oauth/google";
import App from "../../components/App";

describe("App", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <GoogleOAuthProvider>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </GoogleOAuthProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders 'Surreal Estate'", () => {
    render(
      <GoogleOAuthProvider>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </GoogleOAuthProvider>
    );

    const appHeader = screen.getByText("Surreal Estate");

    expect(appHeader).toBeInTheDocument();
  });

  it("renders logo", () => {
    render(
      <GoogleOAuthProvider>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </GoogleOAuthProvider>
    );

    const logo = screen.getByAltText("Logo");

    expect(logo).toBeInTheDocument();
  });
});
