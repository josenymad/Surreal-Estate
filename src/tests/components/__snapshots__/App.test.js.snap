// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      class="App__header"
    >
      Surreal 
      <img
        alt="Logo"
        class="App__header__logo"
        src="logo.jpeg"
      />
       Estate
    </h1>
    <div
      class="navbar"
    >
      <ul
        class="navbar__links"
      >
        <li
          class="navbar__links__item"
        >
          <a
            href="/"
          >
            View all Properties
          </a>
        </li>
        <li
          class="navbar__links__item"
        >
          <a
            href="/add-property"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </div>
    <div
      class="properties"
    >
      <h3
        class="properties__header"
      >
        Properties Page
      </h3>
      <div
        class="city-filter"
      >
        <div
          class="filter-sort-search"
        >
          <a
            class="city-filter__link"
            href="/?query={\\"city\\":\\"Manchester\\"}"
          >
            Manchester
          </a>
          <a
            class="city-filter__link"
            href="/?query={\\"city\\":\\"Leeds\\"}"
          >
            Leeds
          </a>
          <a
            class="city-filter__link"
            href="/?query={\\"city\\":\\"Sheffield\\"}"
          >
            Sheffield
          </a>
          <a
            class="city-filter__link"
            href="/?query={\\"city\\":\\"Liverpool\\"}"
          >
            Liverpool
          </a>
          <a
            class="city-filter__link"
            href="/?sort={\\"price\\":1}"
          >
            Low to High
          </a>
          <a
            class="city-filter__link"
            href="/?sort={\\"price\\":-1}"
          >
            High to Low
          </a>
        </div>
      </div>
      <p
        class="no-properties"
      >
        Sorry, there are no available properties in 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
